healthcheck.go

```diff
@@ -0,0 +1,47 @@
+package client
+
+import (
+	"context"
+	health "github.com/ONSdigital/dp-healthcheck/healthcheck"
+	"github.com/ONSdigital/log.go/log"
+	"net/http"
+)
+
+// ServiceName
+const ServiceName = "Nomad"
+
+var (
+	// StatusMessage contains a map of messages to service response statuses
+	StatusMessage = map[string]string{
+		health.StatusOK:       " is ok",
+		health.StatusWarning:  " is degraded, but at least partially functioning",
+		health.StatusCritical: " functionality is unavailable or non-functioning",
+	}
+)
+
+func (c *Client) Checker(ctx context.Context, state *health.CheckState) error {
+	service := ServiceName
+	logData := log.Data{
+		"service": service,
+	}
+
+	code, err := c.Get(ctx, "/v1/agent/health?type=client")
+	if err != nil {
+		log.Event(ctx, "failed to request service health", log.ERROR, log.Error(err), logData)
+		return err
+	}
+
+	if code != http.StatusOK {
+		message := generateMessage(service, health.StatusCritical)
+		state.Update(health.StatusCritical, message, code)
+	} else {
```

Comment:
Very minor but i would suggest not using an else here, and making the success case the primary line of sight of the code. 

So inside the if != OK, return nil, then just remove the else condition and make the state.Update(ok) stuff unindented

Reply:
This is because idiomatic Go considers the least indented line of code to be 'happy path' so we try and keep successful conditions at that level https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88